wa/workloads/exoplayer/__init__.py

```diff
@@ -0,0 +1,185 @@
+                  description='Specifies which format video file to play.'),
+        Parameter('filename',
+                  description="""
+                   The name of the video file to play. This can be either a path
+                   to the file anywhere on your file system, or it could be just a
+                   name, in which case, the workload will look for it in
+                   ``{}``
+                   *Note*: either format or filename should be specified, but not both!
+                  """.format(video_directory)),
+        Parameter('force_dependency_push', kind=boolean, default=False,
+                  description="""
+                  If true, video will always be pushed to device, regardless
+                  of whether the file is already on the device.  Default is ``False``.
+                  """),
+    ]
+
+    @property
+    def host_video_file(self):
+        if not self._selected_file:
+            if self.filename:
+                if self.filename[0] in './' or len(self.filename) > 1 and self.filename[1] == ':':
+                    filepath = os.path.abspath(self.filename)
+                else:
+                    filepath = os.path.join(self.video_directory, self.filename)
+                if not os.path.isfile(filepath):
+                    raise WorkloadError('{} does not exist.'.format(filepath))
+                self._selected_file = filepath
+            else:
+                files = self.video_files[self.format]
+                if not files:
+                    url = DOWNLOAD_URLS[self.format]
+                    filepath = os.path.join(self.video_directory, os.path.basename(url))
+                    self.logger.debug('Downloading {}...'.format(filepath))
+                    urllib.urlretrieve(url, filepath)
+                    self._selected_file = filepath
+                else:
+                    self._selected_file = files[0]
+                    if len(files) > 1:
+                        self.logger.warn('Multiple files for 720p found. Using {}.'
+                                         .format(self._selected_file))
+                        self.logger.warn('Use "filename"parameter instead of '
+                                         '"format" to specify a different file.')
+        return self._selected_file
+
+    def init_resources(self, context):
+        ensure_directory_exists(self.video_directory)
+        self.video_files = defaultdict(list)
+        self.enum_video_files()
+        self._selected_file = None
+
+    def enum_video_files(self):
+        for filename in os.listdir(self.video_directory):
+            for format in self.parameters['format'].allowed_values:
+                if format in filename:
+                    self.video_files[format].append(os.path.join(self.video_directory, filename))
+
+    def validate(self):
+        if (self.format and self.filename) and (self.format != self.parameters['format'].default):
```

Comment:
What if `format` explicitly set to the default value along side a filename in the config? Should default it to `None` and then set it to default here after the first check.